shieldlib/src/shield_renderer.ts

```diff
@@ -52,15 +56,27 @@ export type ShapeDrawFunction = (
 ) => number;
 
 class MaplibreGLSpriteRepository implements SpriteRepository {
-  map: Map;
-  constructor(map: Map) {
+  map: MapLibre;
+  constructor(map: MapLibre) {
     this.map = map;
   }
   getSprite(spriteID: string): StyleImage {
     return this.map.style.getImage(spriteID);
   }
-  putSprite(spriteID: string, image: ImageData, pixelRatio: number): void {
-    this.map.addImage(spriteID, image, { pixelRatio: pixelRatio });
+  putSprite(
+    spriteID: string,
+    image: ImageData,
+    pixelRatio: number,
+    update: boolean
+  ): void {
+    if (update) {
+      console.log(`update ${spriteID}`);
+      this.map.removeImage(spriteID);
+      this.map.addImage(spriteID, image);
+    } else {
+      console.log(`add ${spriteID}`);
+      this.map.addImage(spriteID, image, { pixelRatio: pixelRatio });
```

Comment:
The last parameter to `addImage()` accepts [some other options](https://maplibre.org/maplibre-gl-js/docs/API/types/maplibregl.StyleImageMetadata/), of which `sdf` may be the most relevant to a client of this library. Should we plumb through an entire options object?